protocol/storage/client.go

```diff
@@ -178,6 +197,26 @@ func (smc *Client) ProposeDeal(ctx context.Context, miner address.Address, data
 	return &response, nil
 }
 
+func (smc *Client) pingMiner(ctx context.Context, pid peer.ID) error {
+	ctx, cancel := context.WithTimeout(ctx, 15*time.Second) //TODO: better place to (not) hard-code this?
```

Comment:
For now, please make the timeout a parameter to this function, then move this hardcoded value next to the other hardcoded value at line 90. Remove the TODO before merging.